File: 2024/day3/day3.ts
import { argv } from 'process';

import {readFileSync} from "fs";
import {Result} from "../shared/types.js"

const INPUT_DATA = "day3/input.txt";
export function day3(){
    let res: Result = {p1: "", p2: ""};
    const input = readFileSync(INPUT_DATA, {encoding: "utf-8"});
    res.p1 = "0";
    res.p2 = "0";
    return res;
}

if (import.meta.filename === argv[1]) {
    const result = day3();
    console.log(`Day 3 Part 1: ${result.p1}`);
    console.log(`Day 3 Part 2: ${result.p2}`);
} 

File: 2024/day3/day3.py
from shared.types import Result
from shared.input import get_input
from shared.parser.interpret import interpret_tokens
from shared.parser.tokenizer import Tokenizer
import re

INPUT_DATA = "day3/input.txt"
SAMPLE_DATA = "day3/sample.txt"


def process_instructions(data, regex: re.Pattern):
    total = 0
    active = True
    hits = regex.findall(data)
    for hit in hits:
        if hit == 'do()':
            active = True
        elif hit == "don't()":
            active = False
        elif active:
            digits = hit[4:-1].split(',')
            total += int(digits[0]) * int(digits[1])
    return total


def day3() -> Result:
    res = Result()
    input = ''.join(get_input(INPUT_DATA))
    tokenizer = Tokenizer(input)
    res.p1 = interpret_tokens(tokenizer)
    tokenizer.reset()
    res.p2 = interpret_tokens(tokenizer, True)
    return res


if __name__ == '__main__':
    result = day3()
    print(f'Day 3 Part 1: {result.p1}')
    print(f'Day 3 Part 2: {result.p2}')
